rust/index/src/lib.rs

```diff
@@ -8,5 +8,21 @@ pub mod utils;
 
 // Re-export types
 
+use chroma_cache::new_non_persistent_cache_for_test;
+use chroma_storage::test_storage;
 pub use hnsw::*;
+use hnsw_provider::HnswIndexProvider;
+use tempfile::tempdir;
 pub use types::*;
+
+pub fn test_hnsw_index_provider() -> HnswIndexProvider {
```

Comment:
nit: maybe gate with `#[cfg(test)]`

Reply:
Discussed offline: we also need to make this available for benchmark, but `#[cfg(test)]` will not work with benchmark out of the box